util.c

```diff
@@ -787,3 +787,65 @@ char *filter_printf(const char *filter, const char *user) {
   filter_result_len(filter, user, result);
   return result;
 }
+
+size_t rfc4515_length ( const char* in, char* out )
+{
+    const char* pos = NULL;
+    size_t result = 0;
+    char* specials = (char*)malloc( 4 * sizeof(char) );
+    specials[0] = 0x5c;
+    specials[1] = 0x2a;
+    specials[2] = 0x28;
+    specials[3] = 0x29;
+
+    do
+    {
+        pos = strpbrk( in, specials );
+
+        size_t len = 0;
+        if( pos != NULL )
+        {
+            len = pos - in;
+        }
+        else
+        {
+            len = strlen( in );
+        }
+
+        if( out != NULL )
+        {
+            strncpy( out, in, len );
+            out += len;
+        }
+
+        result += len;
+        in += len;
+
+        if( pos != NULL )
+        {
+            if( out != NULL )
+            {
+              *(out) = 0x5c;
+              strncpy( (out+1), pos, 1 );
+              out += 2;
+            }
+
+            in += sizeof(char);
+            result += 2;
+        }
+    } while ( pos != NULL );
+
+    if( out != NULL )
+    {
+        *out = '\0';
+    }
+
+    free ( specials );
+    return ( result + sizeof( char ) );}
+
+char* rfc4515_replace ( const char* in )
+{
```

Comment:
I think I'd like us to skip this function, I like the pattern where you're responsible for the memory you allocate. I much prefer the two function calls where needed rather than keeping track that this memory needs to be freed.

Reply:
Hey @klali, I am not sure what du you mean. I need to ask, because english is not my native language. Do you prefer removing rfc4515_replace(...) and call rfc4515_length(...) in the function in which the username is required?

In example: The user name is passed to filter_printf(...). In filter_printf(...) my function rfc4515_length(...) is called to determine the length of the "new" user name and to allocate the memory. Then rfc4515_length(...) is called again to write the "new" user name to the allocated memory. After using the "new" user name, the allocated memory is released in filter_printf(...).

Reply:
Yeah.. I see what you mean with the pattern we have. I would've liked to not have the extra function just for allocating memory, but it's fine. This project is a mess.